src/adm.rs

```diff
@@ -1,39 +1,269 @@
 use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, HashMap};
 use url::Url;
 
-use crate::error::HandlerError;
+use crate::error::{HandlerError, HandlerErrorKind, HandlerResult};
+use crate::server::location::LocationResult;
+use crate::server::ServerState;
+use crate::settings::Settings;
+use crate::tags::Tags;
+use crate::web::middleware::sentry as l_sentry;
+//use crate::server::img_storage;
```

Comment:
Nit: should this commented out use statement be here?

Reply:
ah, that's probably a hold over from the image caching PR. We're still debating if it's needed or if we can push straight to the CDN.